beangulp/importers/csvbase.py

```diff
@@ -137,17 +140,24 @@ class Amount(Column):
       subs: Dictionary mapping regular expression patterns to
         replacement strings. Substitutions are performed with
         re.sub() in the order they are specified.
+      negate: If true, flip the sign of the amount.
 
     """
 
-    def __init__(self, name, subs=None):
+    def __init__(self, name, subs=None, negate=False):
         super().__init__(name)
         self.subs = subs if subs is not None else {}
+        self.negate = negate
 
     def parse(self, value):
+        if value == "":
+            return None
```

Comment:
Why?

Reply:
Please note that the `Column` base class already has a mechanism for returning a default value when the value is empty. However, it is not exposed by the `Amount` subclass. It probably should. And the mechanism should be extended to make `None` a valid default. However, making returning `None` on empty values the default behavior of the `Amount` column description does not seem to be a good idea to me.

Reply:
A CSV file can have multiple Amount columns, e.g. one for "DEBIT" and another for "CREDIT". Only one of the two has value and the other is an empty string. `Decimal('')` raises exception so user cannot further process the row in `finalize()`. So I'm allowing empty string as Amount here.

Do you think `Decimal('0')` is a better idea than `None`?

Reply:
The default value should be specified by the user.

However, for statements that have a "debit" and a "credit" columns, you most likely want something different, see the code proposed here https://github.com/beancount/beangulp/pull/126#issuecomment-2211999829 I don't remember testing it, but it should work with little adjustments.

Reply:
Thanks. The `WithdrawlOrDeposit` idea is awesome.

Reply:
If it is useful, we can add it to beangulp. The only difficult thing is to find a suitable name. I don't like `WithdrawalOrDeposit`, `DebitOrCredit` is a bit better, but not much. Does anyone have a better idea?

Reply:
`DebitOrCredit` sounds good to me. If you want a new name...

... since we have `Columns`, how about `Amounts`? It takes multiple columns, and finds the non-empty one.

But as `Columns` concats strings together, `Amounts` might make people think that it adds up decimals together.

Reply:
Added in #155.